apps/aehyperchains/test/aehc_contract_tests.erl

```diff
@@ -0,0 +1,702 @@
+-module(aehc_contract_tests).
+
+
+-include("../../aecontract/src/aect_sophia.hrl").
+-include("../../aecontract/include/aecontract.hrl").
+-include("../../aecontract/test/include/aect_sophia_vsn.hrl").
+-include_lib("aecontract/include/hard_forks.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+
+-define(cid(__x__), {'@ct', __x__}).
+-define(hsh(__x__), {'#', __x__}).
+-define(sig(__x__), {'$sg', __x__}).
+-define(oid(__x__), {'@ok', __x__}).
+-define(qid(__x__), {'@oq', __x__}).
+
+-define(ALOT, (1000000000000000000000000000000000000000000000000 * aec_test_utils:min_gas_price())).
+-define(CALL_COST, (1000 * aec_test_utils:min_gas_price())).
+-define(CALL_GAS, 1000000).
+
+
+setup_log_file() ->
+    LogFileDir = filename:join(
+        hd(string:split(code:priv_dir(aesophia), "_build", trailing)),
+        "_build/test/logs/"),
+    LogFileName =
+        begin
+            {{Year,Month,Day},{Hour,Min,Sec}} = erlang:localtime(),
+            io_lib:format("chain_simulator_~p-~p-~p_~p:~p:~p.log", [Year, Month, Day, Hour, Min, Sec])
+        end,
+    LogFilePath = filename:join(LogFileDir, LogFileName),
+    filelib:ensure_dir(LogFilePath),
+    {ok, LogFile} = file:open(LogFilePath, [write]),
+    LogFile.
+
+setup() ->
+    erlang:system_flag(backtrace_depth, 100),
+    application:ensure_all_started(gproc),
+    aec_test_utils:mock_genesis_and_forks(),
+    mock_protocol(),
+    ok = lager:start(),
+    group_leader(setup_log_file(), self()),
+    aec_keys:start_link(),
+    {ok, _} = aec_chain_sim:start(),
+    ok.
+
+unsetup(ok) ->
+    aec_chain_sim:stop(),
+    unmock_protocol(),
+    aec_test_utils:unmock_genesis_and_forks().
+
+
+mock_protocol() ->
```

Comment:
This means that those tests are to be ran on `iris` for all eunit runs (ex. when `make eunit-roma` is executed), not a bug but rather unexpected. Also it has to be revised once a new protocol name is designed - the one after `iris`.

Reply:
Do you have some idea how to make it more general?